apps/web/src/components/SkillDialog/SkillSelection.tsx

```diff
@@ -0,0 +1,275 @@
+import React from "react";
+import { useFormContext } from "react-hook-form";
+import { useIntl } from "react-intl";
+
+import {
+  Button,
+  Collapsible,
+  Link,
+  Separator,
+  Well,
+} from "@gc-digital-talent/ui";
+import { Combobox, Select } from "@gc-digital-talent/forms";
+import { errorMessages, getLocalizedName } from "@gc-digital-talent/i18n";
+
+import { invertSkillSkillFamilyTree } from "~/utils/skillUtils";
+import { Skill, SkillCategory } from "~/api/generated";
+import ChevronRightIcon from "@heroicons/react/20/solid/ChevronRightIcon";
+import SkillDescription from "./SkillDescription";
+import useRoutes from "../../hooks/useRoutes";
+
+const suggestionLink = (chunks: React.ReactNode, href: string) => (
+  <Link href={href}>{chunks}</Link>
+);
+
+interface SkillSelectionProps {
+  skills: Skill[];
+  onSelectSkill?: (skill: Skill | null) => void;
+  showCategory?: boolean;
+}
+
+const SkillSelection = ({
+  skills,
+  onSelectSkill,
+  showCategory = true,
+}: SkillSelectionProps) => {
+  const intl = useIntl();
+  const paths = useRoutes();
+  const [isExpanded, setIsExpanded] = React.useState<boolean>(false);
+  const { watch, resetField } = useFormContext();
+
+  const [category, family, skill] = watch(["category", "family", "skill"]);
+
+  const allSkillFamilies = React.useMemo(
+    () => invertSkillSkillFamilyTree(skills),
+    [skills],
+  );
+
+  const filteredSkills = React.useMemo(() => {
+    if (family) {
+      // We only care about family if it is set
+      // since we are filtering families by category
+      return skills.filter((currentSkill) =>
+        currentSkill.families?.some((skillFamily) => skillFamily.id === family),
+      );
+    }
+    if (category) {
+      return skills.filter((currentSkill) =>
+        currentSkill.families?.some(
+          (skillFamily) => skillFamily.category === category,
+        ),
+      );
+    }
+
+    // neither is set so return all skills
+    return skills;
+  }, [category, family, skills]);
+
+  const selectedSkill = React.useMemo(() => {
+    return skill
+      ? skills.find((currentSkill) => currentSkill.id === skill)
+      : undefined;
+  }, [skill, skills]);
+
+  React.useEffect(() => {
+    if (onSelectSkill) {
+      onSelectSkill(selectedSkill || null);
+    }
+  }, [onSelectSkill, selectedSkill]);
+
+  React.useEffect(() => {
+    resetField("skill");
+  }, [category, family, resetField]);
+
+  return (
+    <>
+      <div
+        data-h2-display="base(grid)"
+        data-h2-gap="base(x1)"
+        data-h2-margin-bottom="base(x1)"
+        {...(showCategory
+          ? {
+              "data-h2-grid-template-columns": "base(1fr) p-tablet(1fr 1fr)",
+            }
+          : {
+              "data-h2-grid-template-columns": "base(1fr)",
+            })}
+      >
+        {showCategory && (
+          <Select
```

Comment:
It seems like the _category_ selection has no effect if a skill family is selected.  They are not "anding" or "oring".  Maybe the _category_ filter could be reset to blank when a skill family is selected to show what is happenign?

Reply:
Oh, this is a good catch. I think I completely messed this up 😓 @substrae are you able to confirm what is intended to happen when a user changes either filter?

Reply:
@esizer @petertgiles 

To be honest I'm not sure how skill families and categories interact with each other, but the way I expected this to work was that filtering by category would only show the skill families in that category, and only the skills in that category. Selecting a skill family and all categories and then choosing a narrower category would reset skill families in the event that the category didn't contain the existing skill family.

Reply:
Oh yeah, this is nice.  A+.